openstf/tasks/utils/predictionjobloop.py

```diff
@@ -138,6 +140,7 @@ def map(self, function, *args, **kwargs):
 
             except Exception as exception:
                 pids_unsuccessful.append(prediction_job["id"])
```

Comment:
I would suggest to not strip information here and use a single datastructure:

```python
# We can always remove information later if we want, but for now keep the full exception I guess (minor)
pids_unsucessful[prediction_job["id"]] = exception
```

Reply:
Now I see what you try to achieve grouping the exceptions. I can see some benefits to doing that. Still I would recommend to use a single data structure. Perhaps `pids_exceptions` is more clean in this case.

Reply:
I agree that the list of pids_unsecessful is a bit superfuous af we move to a dict with {exceptions : pid-list}. I think your code results in something different from both those options, right?

Reply:
I think we mean the same. Probably a dict with exception messages as keys and list of pid as values is the way to go.